src/dmd/root/array.d

```diff
@@ -138,8 +138,11 @@ public:
     void reserve(size_t nentries) pure nothrow
     {
         //printf("Array::reserve: length = %d, data.length = %d, nentries = %d\n", (int)length, (int)data.length, (int)nentries);
-        if (data.length - length < nentries)
+
+        // Cold path
+        void enlarge(size_t nentries)
```

Comment:
Doesn't  this impose some penalty for being a nested function?

Reply:
Non-static nested functions have an extra parameter which is a pointer to the enclosing stack frame.

But it doesn't matter here. Calling `enlarge` should be rare, which is the point.